### PR TITLE
[bug] [ir] Fix Block::replace_with segment faults when stmts.size() == 0

### LINKED ISSUE #1071: [Lang] print fails on a simple test case
**Describe the bug**
```python
import taichi as ti
ti.init(print_ir=True)

@ti.kernel
def fill():
    for i in range(1):
        print(i)

fill()
```

**Log/Screenshots**
```
[Taichi] mode=development
[Taichi] preparing sandbox at /tmp/taichi-q5q3s29d
[Taichi] sandbox prepared
[Taichi] <dev mode>, supported archs: [cpu, cuda, opengl], commit b4b68529, python 3.6.9
[I 05/28/20 22:00:46.405] [compile_to_offloads.cpp:operator()@21] Initial IR:
kernel {
  $0 : for @tmp0 in range((cast_value<int32> 0), (cast_value<int32> 1)) {
    $1 = eval @tmp0
    print %1
  }
}
[E 05/28/20 22:00:46.405] Received signal 11 (Segmentation fault)


***********************************
* Taichi Compiler Stack Traceback *
***********************************
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::Logger::error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::signal_handler(int)
/lib/x86_64-linux-gnu/libc.so.6(+0x3ef20) [0x7f7907462f20]
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::lang::Block::replace_with(taichi::lang::Stmt*, taichi::lang::VecStatement&&, bool)
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::lang::LowerAST::visit(taichi::lang::FrontendEvalStmt*)
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::lang::LowerAST::visit(taichi::lang::Block*)
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::lang::LowerAST::visit(taichi::lang::RangeForStmt*)
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::lang::LowerAST::visit(taichi::lang::Block*)
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::lang::LowerAST::run(taichi::lang::IRNode*)
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::lang::irpass::compile_to_offloads(taichi::lang::IRNode*, taichi::lang::CompileConfig const&, bool, bool, bool, bool, bool)
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::lang::Kernel::lower(bool)
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::lang::Program::compile(taichi::lang::Kernel&)
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::lang::Kernel::compile()
/tmp/taichi-q5q3s29d/taichi_core.so: taichi::lang::Kernel::operator()()
/tmp/taichi-q5q3s29d/taichi_core.so(+0x70fed4) [0x7f78eca8aed4]
/tmp/taichi-q5q3s29d/taichi_core.so(+0x66998f) [0x7f78ec9e498f]
python3.6(_PyCFunction_FastCallDict+0x154) [0x55ae1509dc54]
python3.6(_PyObject_FastCallDict+0x2bf) [0x55ae1509e06f]
python3.6(_PyObject_Call_Prepend+0x63) [0x55ae150a2aa3]
python3.6(PyObject_Call+0x3e) [0x55ae1509da5e]
python3.6(+0x16b371) [0x55ae150f7371]
python3.6(_PyObject_FastCallDict+0x8b) [0x55ae1509de3b]
python3.6(+0x199c0e) [0x55ae15125c0e]
python3.6(_PyEval_EvalFrameDefault+0x30a) [0x55ae1514875a]
python3.6(PyEval_EvalCodeEx+0x966) [0x55ae15120ff6]
python3.6(+0x1957d4) [0x55ae151217d4]
python3.6(PyObject_Call+0x3e) [0x55ae1509da5e]
python3.6(_PyEval_EvalFrameDefault+0x19e7) [0x55ae15149e37]
python3.6(+0x192e66) [0x55ae1511ee66]
python3.6(_PyFunction_FastCallDict+0x3d8) [0x55ae15120598]
python3.6(_PyObject_FastCallDict+0x26f) [0x55ae1509e01f]
python3.6(_PyObject_Call_Prepend+0x63) [0x55ae150a2aa3]
python3.6(PyObject_Call+0x3e) [0x55ae1509da5e]
python3.6(+0x16b371) [0x55ae150f7371]
python3.6(PyObject_Call+0x3e) [0x55ae1509da5e]
python3.6(_PyEval_EvalFrameDefault+0x19e7) [0x55ae15149e37]
python3.6(+0x193136) [0x55ae1511f136]
python3.6(+0x193ed6) [0x55ae1511fed6]
python3.6(+0x199b95) [0x55ae15125b95]
python3.6(_PyEval_EvalFrameDefault+0x30a) [0x55ae1514875a]
python3.6(PyEval_EvalCodeEx+0x329) [0x55ae151209b9]
python3.6(PyEval_EvalCode+0x1c) [0x55ae1512175c]
python3.6(+0x215744) [0x55ae151a1744]
python3.6(PyRun_FileExFlags+0xa1) [0x55ae151a1b41]
python3.6(PyRun_SimpleFileExFlags+0x1c3) [0x55ae151a1d43]
python3.6(Py_Main+0x613) [0x55ae151a5833]
python3.6(main+0xee) [0x55ae1506f88e]
/lib/x86_64-linux-gnu/libc.so.6: __libc_start_main
python3.6(+0x1c3160) [0x55ae1514f160]
```

### PR DESCRIPTION
**Pull request by archibate**
Related issue = close #1071

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

Seems `replace_usage = true` not working when `new_statements.size() == 0`:
https://github.com/taichi-dev/taichi/blob/c193e4b431d52ffa4e875940ece589d5be9599f0/taichi/ir/ir.cpp#L760-L761
... where `new_statements.back() == nullptr`.

### CODE REVIEW

File: taichi/transforms/lower_ast.cpp [hunk 1]
```diff
@@ -331,7 +331,10 @@ class LowerAST : public IRVisitor {
     auto fctx = make_flatten_ctx();
     expr->flatten(&fctx);
     stmt->eval_expr.cast<EvalExpression>()->stmt_ptr = stmt->expr->stmt;
-    stmt->parent->replace_with(stmt, std::move(fctx.stmts));
+    if (fctx.stmts.stmts.size())
+      stmt->parent->replace_with(stmt, std::move(fctx.stmts));
+    else
+      stmt->parent->erase(stmt);
```
Comment (archibate) [inline taichi/transforms/lower_ast.cpp:337]: Am I expected to intergrate `erase` into `replace_with`?

**Comment by archibate** [comment]
Hello? @yuanming-hu I know you're busy preparing GAMES201, so can I merge this omo once @xumingkuan approved?

**Comment by yuanming-hu** [comment]
> Hello? @yuanming-hu I know you're busy preparing GAMES201, so can I merge this omo once @xumingkuan approved? There are now 14 PRs waiting for merge, I'll merge some with trival changeset omo with 1 approve from taichi-dev write-members.

Sure, having Mingkuan's approval is good enough. Btw, please capitalize the first letter of your PR letter and avoid confusing abbreviations. Thank you!

**Comment by xumingkuan** [comment]
(Actually we can add `VecStatement::empty()` for convenience.)

**Pull request merged**